### PR TITLE
Refactor path checking code of the loaders

### LINKED ISSUE #91: Add a check_path method to base class Loader
Loaders have redundant code that checks paths, we can add a method to the base class `Loader` to do this

```python
if not isinstance(path, (str, os.PathLike)):
    # In Python 3.8, this can be done with isinstance(path, typing.get_args(_typing.PathLike))
    raise TypeError(f'Unsupported path type "{type(path)}".')
```

### PR DESCRIPTION
**Pull request by edwardleardi**
Move the path checking code to a method in the Loader base class

---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #91 

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [ ] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: pydax/loaders/_base.py [hunk 1]
```diff
@@ -34,3 +34,8 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
         :return: The object representing the loaded file.
         """
         pass
+
+    def check_path(self, path: _typing.PathLike) -> bool:
```
Comment (xuhdev) [inline pydax/loaders/_base.py]:
```suggestion
    def check_path(self, path: _typing.PathLike) -> None:
```

File: pydax/loaders/_base.py [hunk 2]
```diff
@@ -34,3 +34,14 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
         :return: The object representing the loaded file.
         """
         pass
+
+    def check_path(self, path: Union[_typing.PathLike, Dict[str, str]]) -> None:
```
Comment (xuhdev) [inline pydax/loaders/_base.py:38]: I would call this function in the abstract `load()` and let overriding methods call `super().load()`.

File: pydax/loaders/_text.py [hunk 3]
```diff
@@ -34,9 +33,8 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
...
 
-        if not isinstance(path, (str, os.PathLike)):
-            # In Python 3.8, this can be done with isinstance(path, typing.get_args(_typing.PathLike))
-            raise TypeError(f'Unsupported path type "{type(path)}".')
+        self.check_path(path)
 
         encoding = options.get('encoding', 'utf-8')
-        return pathlib.Path(path).read_text(encoding=encoding)
+        # We can remove the error suppression once Dict[str, str] handling is added
+        return pathlib.Path(path).read_text(encoding=encoding)  # type: ignore [arg-type]
```
Comment (xuhdev) [inline pydax/loaders/_text.py]: You can use `typing.cast` to avoid ignoring the error

Comment (edwardleardi) [reply pydax/loaders/_text.py]: got it thanks!

**Review by xuhdev — APPROVED** [review]

**Pull request merged**